kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryption.java

```diff
@@ -105,7 +111,7 @@ private static <K, E> Kms<K, E> buildKms(FilterFactoryContext context, Config co
         kms = InstrumentedKms.wrap(kms, kmsMetrics);
         ExponentialJitterBackoffStrategy backoffStrategy = new ExponentialJitterBackoffStrategy(Duration.ofMillis(500), Duration.ofSeconds(5), 2d,
                 ThreadLocalRandom.current());
-        kms = ResilientKms.wrap(kms, context.eventLoop(), backoffStrategy, 3);
+        kms = ResilientKms.wrap(kms, RETRY_POOL, backoffStrategy, 3);
```

Comment:
I'm a little wary of this fix as it just throws yet another thread into the picture. 

The other fix I'd see is to inject `Supplier<ScheduledExecutorService>` and use that to get an executor when needed as by the time we need to submit we know the context will actually have an eventloop.

Reply:
If you wanted the supplier to get at the eventloop of the filter currently using the Kms that's a bit tricky. ResilientKms is being created once from the main method.

The DekManager and through it the ResilientKms can be used by Filter instances associated with different eventloops.

I don't know if the Supplier could get at the eventloop it's being called from.

Reply:
Another idea is at initialisation time we could pass the server EventLoopGroup through the context to the FilterFactory. That would at least let you execute on _a_ netty eventloop, but maybe not the one that is invoking the retry, we wouldn't be adding new threads at least. It would require updating the FilterFactory APIs

Reply:
We may also be well served by introducing a blocking/work pool for Filters that need to integrate with blocking code or other utilities like this.

Reply:
> The DekManager and through it the ResilientKms can be used by Filter instances associated with different eventloops.

Ohh right thats a nuance I hadn't really considered. I was just working along the lines of restoring the intended behaviour of binding it to the event loop from the construction context. If we want to execute the retries on the filter thread we would need to supply it on every call to the KMS (I think). 

We really need to do some profiling of all this.

Reply:
Maybe we could ditch the resilientKMS and instead have each EnvelopeEncryptionFilter wrap the DekManager in a ResilientDekManager and do the retries up at that level. Having a `ResilientDekManager` per filter wrapped around the shared `DekManager` wouldn't be much overhead.

Still thinking I will merge this as is and look at these better fixes for 0.6.0

Reply:
> have each EnvelopeEncryptionFilter wrap the DekManager in a ResilientDekManager and do the retries up at that level. Having a `ResilientDekManager` per filter wrapped around the shared `DekManager` wouldn't be much overhead.

This idea of replacing the `Kms` decorators with equivalent `DekManager` decorators was something we mentioned when adding `DekManager`. It's definitely something we should look into.